File: 0048-rotate-image/0048-rotate-image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int n = matrix.size();
        int m = matrix[0].size();
        vector<vector<int>> v(n , vector<int> (m , -1));
         int col = n-1;
        for(int i=0;i<n ; i++){
            int row = 0;
            for(int j =0; j<m;j++){
                v[row][col]=matrix[i][j];
                row++;
            }
            col--;
        }
        for(int i=0;i<n;i++){
            for(int j =0; j < m; j++){
                matrix[i][j] = v[i][j];
            }
            // cout<<"\n";
        }
    }
};

File: 0048-rotate-image/0048-rotate-image.py
class Solution:
    def rotate(self, arr: List[List[int]]) -> None:
            n = len(arr)

            for i in range(n):
                for j in range(i+1,n):
                    arr[i][j], arr[j][i] = arr[j][i], arr[i][j]
            for i in range(n):
                arr[i].reverse()
            return arr